core/src/main/scala/cats/mtl/Handle.scala

```diff
@@ -218,5 +219,40 @@ private[mtl] trait HandleInstances extends HandleLowPriorityInstances {
 }
 
 object Handle extends HandleInstances {
+
   def apply[F[_], E](implicit ev: Handle[F, E]): Handle[F, E] = ev
+
+  def ensure[F[_], E]: AdHocSyntax[F, E] =
+    new AdHocSyntax[F, E]
+
+  final class AdHocSyntax[F[_], E] {
+
+    def apply[A](body: Handle[F, E] => F[A])(implicit F: ApplicativeThrow[F]): Inner[A] =
```

Comment:
Could we implement this with `Handle[F, Throwable]` instead of `ApplicativeThrow[F]`?